D1673/P1673.bs

```diff
@@ -3947,6 +4188,19 @@ void triangular_matrix_matrix_right_solve(
   Triangle t,
   DiagonalStorage d,
   inout_matrix_t B);
+
+template<class T>
+T <it>conj-if-needed</it>(const T& t) // <it>exposition only</it>
+{
+    using std::conj;
+    return conj(t);
+}
```

Comment:
I'm stretching here, but should `std::blas` support types whose conjugate is not the same type? As written, this implicitly says "no".

I'm struggling to come up with good examples other than dependent-type trickery (e.g. polar representations whose types limit their `theta`), but that feels like my mathematical ignorance, not fundamental.

It definitely seems reasonable to me to say that computational efficiency demands this equivalence, though.

Reply:
> I'm stretching here, but should std::blas support types whose conjugate is not the same type? As written, this implicitly says "no".

For all arithmetic types, `conj(x)` is `complex<double>`.  The point of _`conj-if-needed`_ is to fix that weird thing, so that (for the sake of computational efficiency), `conjugated` preserves the value type of its input.

For (real, imaginary) complex number representations, it would be weird if conjugation behaved differently than unary minus.  For (magnitude, argument) representations, conjugation just rotates by 180 degrees.  In either case, I find it hard to imagine why `conj` should change the type.

Reply:
That being said, the easy fix for types like this would be for _`conj-if-needed`_ to return `auto` instead of `T`.  Thanks for the suggestion!  I'll make this change in PR #242.

Reply:
I've made the change in PR #242 -- some `accessor_conjugate` wording had to change too.  Thanks!